src/clp_ffi_js/ir/StructuredIrUnitHandler.hpp

```diff
@@ -0,0 +1,118 @@
+#ifndef CLP_FFI_JS_IR_STRUCTUREDIRUNITHANDLER_HPP
+#define CLP_FFI_JS_IR_STRUCTUREDIRUNITHANDLER_HPP
+
+#include <cstddef>
+#include <memory>
+#include <optional>
+#include <string>
+#include <utility>
+#include <vector>
+
+#include <clp/ffi/ir_stream/decoding_methods.hpp>
+#include <clp/ffi/KeyValuePairLogEvent.hpp>
+#include <clp/ffi/SchemaTree.hpp>
+#include <clp/ffi/Value.hpp>
+#include <clp/time_types.hpp>
+#include <spdlog/spdlog.h>
+
+#include <clp_ffi_js/constants.hpp>
+#include <clp_ffi_js/ir/LogEventWithFilterData.hpp>
+
+namespace clp_ffi_js::ir {
+using schema_tree_node_id_t = std::optional<clp::ffi::SchemaTree::Node::id_t>;
+
+/**
+ * Class that implements the `clp::ffi::ir_stream::IrUnitHandlerInterface` to buffer log events and
+ * determine the schema-tree node ID of the log level and timestamp kv-pair.
+ */
+class StructuredIrUnitHandler {
+public:
+    /**
+     * @param deserialized_log_events The vector in which to store deserialized log events.
+     * @param log_level_key Key name of schema-tree node that contains the authoritative log level.
+     * @param timestamp_key Key name of schema-tree node that contains the authoritative timestamp.
+     */
+    StructuredIrUnitHandler(
+            std::shared_ptr<std::vector<LogEventWithFilterData<StructuredLogEvent>>>
+                    deserialized_log_events,
+            std::string log_level_key,
+            std::string timestamp_key
+    )
+            : m_log_level_key{std::move(log_level_key)},
+              m_timestamp_key{std::move(timestamp_key)},
+              m_deserialized_log_events{std::move(deserialized_log_events)} {}
+
+    // Methods implementing `clp::ffi::ir_stream::IrUnitHandlerInterface`.
+    /**
+     * Buffers the log event.
+     * @param log_event
+     * @return IRErrorCode::IRErrorCode_Success
+     */
+    [[nodiscard]] auto handle_log_event(StructuredLogEvent&& log_event
+    ) -> clp::ffi::ir_stream::IRErrorCode;
+
+    /**
+     * @param utc_offset_old
+     * @param utc_offset_new
+     * @return IRErrorCode::IRErrorCode_Success
+     */
+    [[nodiscard]] static auto handle_utc_offset_change(
+            [[maybe_unused]] clp::UtcOffset utc_offset_old,
+            [[maybe_unused]] clp::UtcOffset utc_offset_new
+    ) -> clp::ffi::ir_stream::IRErrorCode {
+        SPDLOG_WARN("UTC offset change packets aren't handled currently.");
+
+        return clp::ffi::ir_stream::IRErrorCode::IRErrorCode_Success;
+    }
```

Comment:
@kirkrodrigues Do we have any guidelines about separating the method body from its declaration when we have both `.cpp` + `.hpp` files for the class? If so, shall we move this and also the constructor body to the cpp file?